app/src/main/res/layout/fragment_view_image.xml

```diff
@@ -24,18 +24,54 @@
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
-    <TextView
-        android:id="@+id/mediaDescription"
+    <!--    This should be inside CoordinatorLayout for two reasons:
+
+            1. TouchImageView really wants some constraints ans has no size otherwise
+            2. We don't want sheet to overlap with appbar but the only way to do it with autosizing
+            is to gibe parent some margin. -->
+    <androidx.coordinatorlayout.widget.CoordinatorLayout
```

Comment:
Did you try to have the `CoordinatorLayout` as the top level layout? With `TouchImageView`, `Progressbar` and bottom sheet as childs. Should be possible imho.

Reply:
I tried that first, I left a comment about it in the previous lines. Didn't find a way to prevent overlapping the app bar otherwise.

Reply:
ah I see 🤦 
Why do Android Layouts suck so hard sometimes

Reply:
Yeah it's a question to BottomSheet, you can either set upper limit manually or living auto-sizing decide. It wouldn't add much complexity to expose upper bound IMO